### ISSUE TITLE
Modification of 1st parameter of $append

### ISSUE THREAD

**Issue by HiroyasuNishiyama** [jsonata-js/jsonata#139]
The following code produces ```{ a: [ 1 ] }``` instead of ```{ a: [] }```.

```
var jsonata = require("jsonata");
var data = { a: [] };
var exp = jsonata("$append(a,[1])");
exp.evaluate(data);
console.log(data);
```

The 1st parameter of `$append` is modified as a side effect of `$append` call.
The documentation of [$append](http://docs.jsonata.org/array-functions.html#appendarray1-array2) do not mention this side effect.　Also, similar function for array such as `$reverse`  or `$sort` do not modify their argument.

**Comment by andrew-coleman** [comment]
This is a bug.  Many thanks for reporting.

**Comment by mattbaileyuk** [comment]
Fixed via #142
